### ISSUE TITLE
computing graph layout changes Louvain, label propagation

### ISSUE THREAD

**Issue by EnbyMonkey** [ftheberge/GraphMiningNotebooks#7]
I'm currently working through section 5.9, problem 1. I borrowed code from the provided python notebook (2nd ed). Here is my code:
`
import igraph as ig
from sklearn.metrics import adjusted_mutual_info_score as AMI
import random

random.seed(12345)
G = ig.Graph.Famous('zachary')
G.vs['comm'] = [0,0,0,0,0,0,0,0,1,1,0,0,0,0,1,1,0,0,1,0,1,0,1,1,1,1,1,1,1,1,1,1,1,1]
G['layout'] = G.layout_fruchterman_reingold()

## Leiden
partition = G.community_leiden(objective_function='modularity').membership
print('Leiden: Number of communities:',max(partition)+1, 'AMI:', AMI(G.vs['comm'],partition))

## Louvain
partition = G.community_multilevel().membership
print('Louvain: Number of communities:',max(partition)+1, 'AMI:', AMI(G.vs['comm'],partition))

## Infomap
partition = G.community_infomap().membership
print('Infomap: Number of communities:',max(partition)+1, 'AMI:', AMI(G.vs['comm'],partition))

## LP
partition = G.community_label_propagation().membership
print('Label propagation: Number of communities:',max(partition)+1, 'AMI:', AMI(G.vs['comm'],partition))

## CNM
partition = G.community_fastgreedy().as_clustering().membership
print('CNM: Number of communities:',max(partition)+1, 'AMI:', AMI(G.vs['comm'],partition))
`

The output is
`Leiden: Number of communities: 4 AMI: 0.671159427880139
Louvain: Number of communities: 4 AMI: 0.5653497612707898
Infomap: Number of communities: 3 AMI: 0.6874223479049214
Label propagation: Number of communities: 2 AMI: 0.8334659946350973
CNM: Number of communities: 3 AMI: 0.680764509888601`

However, if I comment out `G['layout'] = G.layout_fruchterman_reingold()` then the output becomes
`Leiden: Number of communities: 4 AMI: 0.671159427880139
Louvain: Number of communities: 4 AMI: 0.671159427880139
Infomap: Number of communities: 3 AMI: 0.6874223479049214
Label propagation: Number of communities: 3 AMI: 0.8184487137442285
CNM: Number of communities: 3 AMI: 0.680764509888601`

This seems like a mistake, since computing a graph layout doesn't affect the graph structure.

**Comment by ftheberge** [comment]
most layout algorithms are randomized, and so are most clustering algorithms. your results are comparable up to "randomization" effects so this looks ok. you can try re-seeding after the layout call, or even before every clustering algorithm, or averaging over several runs.
